gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -1472,30 +1472,34 @@ private Credential getCredential(
       throws IOException, GeneralSecurityException {
     Credential credential = null;
 
+    // In case of impersonating service account, service account scopes need to be expanded in order
+    // to request access token from IAM.
+    ImmutableList<String> scopes =
+        isImpersonatingServiceAccount(config)
+            ? CredentialFactory.CLOUD_SCOPES
```

Comment:
Cloud scopes are too broad, for this, I think.

Reply:
Is there any side effect for being broad tho? This is actually a valid scope to set for many of cross GCP requests based on some request docs.

Reply:
It works, but this is less secure than specifying the only necessary scopes.

Reply:
As resolved offline, update default scope for credential to CLOUD_PLATFORM as best practice.